File: scrimage-core/src/test/scala/com/sksamuel/scrimage/Issue89Test.scala
package com.sksamuel.scrimage

import com.sksamuel.scrimage.nio.JpegWriter
import org.scalatest.{Matchers, WordSpec}

class Issue89Test extends WordSpec with Matchers {

  "Loading image" should {
    "not randomly rotate" in {
      for ( k <- 1 to 10 ) {
        val image = ImmutableImage.fromResource("/issue89.jpg")
        image.width shouldBe 319
        image.height shouldBe 397
        val bytes = image.bytes(new JpegWriter())
        val reread = ImmutableImage(bytes)
        reread.width shouldBe 319
        reread.height shouldBe 397
      }
    }
  }
}

File: scrimage-tests/src/test/kotlin/com/sksamuel/scrimage/core/Issue89Test.kt
package com.sksamuel.scrimage.core

import com.sksamuel.scrimage.ImmutableImage
import com.sksamuel.scrimage.nio.JpegWriter
import io.kotest.core.spec.style.WordSpec
import io.kotest.matchers.shouldBe

class Issue89Test : WordSpec({

   "Loading image" should {
      "not randomly rotate"  {
         repeat(10) {
            val image = ImmutableImage.loader().fromResource("/issue89.jpg")
            image.width shouldBe 319
            image.height shouldBe 397
            val bytes = image.bytes(JpegWriter())
            val reread = ImmutableImage.loader().fromBytes(bytes)
            reread.width shouldBe 319
            reread.height shouldBe 397
         }
      }
   }

})
